File: src/main/kotlin/ru/havlong/cards/exceptions/dto/ErrorDTO.java
package ru.havlong.cards.exceptions.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ErrorDTO {

    private String error;

    @JsonProperty("error_description")
    private String errorDescription;

    public static ErrorDTO of(String error,
                              String errorDescription) {

        return ErrorDTO.builder()
                .error(error)
                .errorDescription(errorDescription)
                .build();
    }
}

File: src/main/kotlin/ru/havlong/cards/transfer/ErrorDTO.kt
package ru.havlong.cards.transfer

/**
 * 20.06.2020
 * ErrorDTO
 *
 * @author Havlong
 * @version v1.0
 */
data class ErrorDTO(
        val error: String,
        val errorDescription: String
) {
    companion object {
        @JvmStatic
        fun of(
                error: String,
                errorDescription: String
        ): ErrorDTO = ErrorDTO(error, errorDescription)
    }
}
